query/workmanager.go

```diff
@@ -94,6 +94,10 @@ type Config struct {
 	// make this configurable to easily mock the worker used during tests.
 	NewWorker func(Peer) Worker
 
+	// OnMaxTries is function closure that's called on max retries on
+	// workers.
+	OnMaxTries func(string)
```

Comment:
think we should instead use a stricter type here (net.Addr) and mention what is being passed to this call-back so that the caller doesnt have to go look at the code to figure out what is being passed to the call-back it is setting. 

The commit message isnt very accurate imo. `OnMaxTries config lets the caller choose what happens when a query goes over the maximum allowed tries` isnt really true - more so: `OnMaxTries` gives the caller access to the peer's address once a maxmum number of retries have been attempted" or something like that

Reply:
it's worth linking the PR that makes use of this new config option in the PR description so that reviewers can understand the context a bit more

Reply:
> think we should instead use a stricter type here (net.Addr) and mention what is being passed to this call-back so that the caller doesnt have to go look at the code to figure out what is being passed to the call-back it is setting.
> 
> The commit message isnt very accurate imo. `OnMaxTries config lets the caller choose what happens when a query goes over the maximum allowed tries` isnt really true - more so: `OnMaxTries` gives the caller access to the peer's address once a maxmum number of retries have been attempted" or something like that

Will address both points.

Reply:
> think we should instead use a stricter type here (net.Addr) and mention what is being passed to this call-back so that the caller doesnt have to go look at the code to figure out what is being passed to the call-back it is setting.

On a second look, I don't think I should address this since all of the existing code in the neutrino codebase is based off of the peer as a string value.

You could modify the `Peer` interface to have a `NetAddr() net.Addr` but I think it's more prudent to modify the entire neutrino code to use `net.Addr` instead of having this exception just for this function.

Reply:
> On a second look, I don't think I should address this since all of the existing code in the neutrino codebase is based off of the peer as a string value.

I dont think we should design off of that. Rather use the proper type here and then in the neutrino code that uses this we can convert to a string. No need to update the Peer interface there, just convert what this call-back provides into a string at call-time